# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/StreamConsumer.h
# Commit: 6d6adcf77b5a
# Full Hash: 6d6adcf77b5a99316ef743323ccf362972f020dd
# Author: Yury Delendik <ydelendik@mozilla.com>
# Date: 2021-10-22 10:00:52
# Regressor Bug: 1545131
# File Overlap Count: 1
# Description:
#   Bug 1545131 - compress wasm alt data entries (r=jandem)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27992
# ==============================================================================

diff -r a6180f7f6296 -r 6d6adcf77b5a js/public/StreamConsumer.h
--- a/js/public/StreamConsumer.h	Thu Oct 21 21:54:52 2021 +0000
+++ b/js/public/StreamConsumer.h	Thu Oct 21 22:08:14 2021 +0000
@@ -49,9 +49,6 @@
  * as when it is later consumed.
  */
 
-using OptimizedEncodingBytes = js::Vector<uint8_t, 0, js::SystemAllocPolicy>;
-using UniqueOptimizedEncodingBytes = js::UniquePtr<OptimizedEncodingBytes>;
-
 class OptimizedEncodingListener {
  protected:
   virtual ~OptimizedEncodingListener() = default;
@@ -64,7 +61,7 @@
 
   // SpiderMonkey may optionally call storeOptimizedEncoding() after it has
   // finished processing a streamed resource.
-  virtual void storeOptimizedEncoding(UniqueOptimizedEncodingBytes bytes) = 0;
+  virtual void storeOptimizedEncoding(const uint8_t* bytes, size_t length) = 0;
 };
 
 class JS_PUBLIC_API StreamConsumer {